src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -1,49 +1,97 @@
 import { Component, OnDestroy } from '@angular/core';
-import { CurrentContentService } from 'src/app/shared/services/current-content.service';
+import { CurrentContentService, EditAction } from 'src/app/shared/services/current-content.service';
 import { ItemData, ItemDataSource } from '../../services/item-datasource.service';
 import { FormBuilder, Validators } from '@angular/forms';
 import { Subscription } from 'rxjs';
 import { filter } from 'rxjs/operators';
 import { FetchError, Fetching, isReady, Ready } from '../../../../shared/helpers/state';
+import { UpdateItemStringService } from '../../http-services/update-item-string.service';
+import { Item } from '../../http-services/get-item-by-id.service';
+import { TOAST_LENGTH } from '../../../../shared/constants/global';
+import { MessageService } from 'primeng/api';
+import { ERROR_MESSAGE } from '../../../../shared/constants/api';
 
 
 @Component({
   selector: 'alg-item-edit',
   templateUrl: './item-edit.component.html',
-  styleUrls: ['./item-edit.component.scss']
+  styleUrls: ['./item-edit.component.scss'],
+  providers: [MessageService]
 })
 export class ItemEditComponent implements OnDestroy {
+  item: Item;
   itemForm = this.formBuilder.group({
     // eslint-disable-next-line @typescript-eslint/unbound-method
     title: ['', [Validators.required, Validators.minLength(3),]],
   });
   itemLoadingState$ = this.itemDataSource.state$;
```

Comment:
Why "loading state" actually?  State includes if it is loading but it is not a loading state.

Reply:
I've used the same name you used in item detail

Reply:
That's a great name then! 😁

Reply:
That's not a great name but I'll check this later.